src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -726,13 +726,36 @@ static int get_pool_offset(sd_dhcp_server *server, be32_t requested_ip) {
         return be32toh(requested_ip & ~server->netmask) - server->pool_offset;
 }
 
+static int prepare_new_lease(DHCPLease **ret_lease, be32_t address, const DHCPClientId *client_id, const uint8_t *chaddr, be32_t gateway, usec_t expiration){
+        _cleanup_free_ DHCPLease *lease = new0(DHCPLease, 1);
```

Comment:
if you do structured initialization anyway below you can use a plain new() instead of new0() here, as you don't need the zero-initialization, the structured initialization will do that for you